### PR TITLE
feat(Filter): UXD-698 change to listbox

### PR DESCRIPTION
**Pull request by allison-c**
### Purpose 🚀
_clear, one sentence description of primary purpose of this PR_

### Notes ✏️

Replaced `<Select />` by `<ListBox />` in `<Filter />` component, only the column dropdown and single-select value dropdown have the search input, because they might be very long, the rule dropdown is usually limited length

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_
- [x] MINOR (backward compatible) change to `@paprika/filter`
- [ ] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/UXD-698-replace-select-by-listbox

### Screenshots 📸
_optional but highly recommended_

### References 🔗
_relevant Jira ticket / GitHub issues_

### CODE REVIEW

**Comment by nahumzs** [comment]
This might be a small bug?

File: packages/Filter/src/components/Item/Item.js [hunk 1]
```diff
@@ -80,31 +81,29 @@ function Item(props) {
...
   }
 
-  function handleChangeSingleSelectFilterValue(event) {
-    onChangeFilter(types.changeTypes.FILTER_VALUE, { id, value: event.target.value });
+  function handleChangeSingleSelectFilterValue(index, options) {
+    onChangeFilter(types.changeTypes.FILTER_VALUE, { id, value: options[index].value });
   }
 
   function renderRuleField() {
     switch (selectedColumnType) {
```
Comment (nahumzs) [inline packages/Filter/src/components/Item/Item.js:93]: curious why to use a switch for two options?

File: packages/Filter/src/components/Item/Item.js [hunk 2]
```diff
@@ -123,9 +122,16 @@ function Item(props) {
...
       case types.columnTypes.BOOLEAN:
         return (
-          <sc.ValueInput as={Select} value={value.toString()} onChange={handleChangeBooleanFilterValue}>
-            <option value="true">{I18n.t("filter.rules.true")}</option>
-            <option value="false">{I18n.t("filter.rules.false")}</option>
+          <sc.ValueInput data-pka-anchor="filter.item.valueInput" value={value.toString()}>
```
Comment (nahumzs) [inline packages/Filter/src/components/Item/Item.js:125]: Each time I review this switch, I want to comment about turning it into an object, but then I realize that each of these statements would end having a different signature if they were turned into a function. I think this is one of the few cases where I find that the switch statement performs better than using an object pattern.

:)

**Review by nahumzs — APPROVED** [review]

File: packages/Filter/src/components/Item/Item.js (hunk 1 above)
Comment (allison-c) [reply packages/Filter/src/components/Item/Item.js:93]: Maybe when I worked on it I thought there might be many column types? To be honest I don't recall, or maybe just a simple mistake 😅

**Pull request merged**